headless-demo/src/jsMain/kotlin/dev/fritz2/headlessdemo/components/combobox.kt

```diff
@@ -0,0 +1,438 @@
+                    opened,
+                    enter = "transition duration-100 ease-out",
+                    enterStart = "opacity-0 scale-95",
+                    enterEnd = "opacity-100 scale-100",
+                    leave = "transition duration-100 ease-in",
+                    leaveStart = "opacity-100 scale-100",
+                    leaveEnd = "opacity-0 scale-95"
+                )
+
+                results.render(into = this) { (query, results, truncated) ->
+                    if (results.isEmpty()) {
+                        p("px-4 py-2 text-gray-400") {
+                            +"No results found"
+                        }
+                    } else {
+                        results.forEach { item ->
+                            comboboxItem(
+                                item,
+                                "w-full relative py-2 pl-10 pr-4 cursor-default select-none disabled:opacity-50 text-sm"
+                            ) {
+                                className(active.map { active->
+                                    if (active) "bg-primary-600 text-white"
+                                    else "text-primary-800"
+                                })
+
+                                highlightedText(item.value.name, query).run {
+                                    className(selected.map {
+                                        if (it) "font-semibold"
+                                        else "font-normal"
+                                    })
+                                }
+
+                                selected.render {
+                                    if (it) {
+                                        span("absolute left-0 inset-y-0 flex items-center pl-3") {
+                                            icon("w-5 h-5", content = HeroIcons.check)
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        if (truncated) {
+                            span("py-2 pl-10 pr-4 text-sm text-gray-400") {
+                                +"Refine your query for more results"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        div("space-y-2") {
+            p("text-sm text-primary-800") {
+                +"Select via data-binding:"
+            }
+            div("flex flex-wrap gap-2") {
+                buildList {
+                    addAll(COUNTRY_LIST.take(2))
+                    addAll(COUNTRY_LIST.takeLast(2))
+                }.forEach { country ->
+                    quickSelectButton(country, selectionStore)
+                }
+            }
+        }
+
+        checkbox("Read-only", readOnlyStore, "checkbox-enable-readonly")
+        checkbox("Auto-select exact matches", enableAutoselectStore, "checkbox-enable-autoselect")
```

Comment:
I suggest to add a hint to try `Oman` to help understanding the consequences visually.
`checkbox("Auto-select exact matches (try 'oman')", enableAutoselectStore, "checkbox-enable-autoselect")`